### PR TITLE
PnL method on AjnaPosition

### PR DESCRIPTION
**Pull request by zerotucks**
## Ticket URL

Please provide a link to ticket:
https://app.shortcut.com/oazo-apps/story/9114/ajna-overview-data-pnl

## Description of Changes

Please list the changes introduced by this PR:

- [x] Add PnL method to AjnaPosition class that access cumulative vals as args

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [x] All checks are passing
- [x] PR is linked to a corresponding ticket
- [x] PR title is clear and concise
- [ ] Code has been self-reviewed and any fixes or improvements noted
- [ ] Documentation has been updated if necessary

### CODE REVIEW

File: packages/dma-library/src/types/ajna/ajna-position.ts [hunk 1]
```diff
@@ -125,6 +132,17 @@ export class AjnaPosition implements IAjnaPosition {
...
+  pnl(
+    cumulativeDepositUSD: BigNumber,
+    cumulativeWithdrawnUSD: BigNumber,
+    cumulativeFeesUSD: BigNumber,
+  ): BigNumber {
+    return this.netValue
+      .minus(cumulativeDepositUSD)
+      .plus(cumulativeWithdrawnUSD)
+      .minus(cumulativeFeesUSD)
+  }
```
Comment (paszkowskiDamian) [inline packages/dma-library/src/types/ajna/ajna-position.ts]:
```export function calculatePNL(events: VaultEvent[], currentNetValueUSD: BigNumber) {
  const cumulativeDepositUSD = events.reduce(getCumulativeDepositUSD, zero)
  const cumulativeWithdrawnUSD = events.reduce(getCumulativeWithdrawnUSD, zero)
  const cumulativeFeesUSD = events.reduce(getCumulativeFeesUSD, zero)

  if (cumulativeDepositUSD.isZero()) {
    return zero
  }

  return cumulativeWithdrawnUSD
    .plus(currentNetValueUSD)
    .minus(cumulativeFeesUSD)
    .minus(cumulativeDepositUSD)
    .div(cumulativeDepositUSD)
}```

We have slightly different formula in borrow. Can we check which one is correct?

File: packages/dma-library/src/types/ajna/ajna-position.ts [hunk 2]
```diff
@@ -125,6 +132,17 @@ export class AjnaPosition implements IAjnaPosition {
       .minus(this.debtAmount.times(this.quotePrice))
   }
 
+  pnl(
```
Comment (piekczyk) [inline packages/dma-library/src/types/ajna/ajna-position.ts:135]: What about taking the same approach as in earn positions? We could potentially calculate it in `getPosition` method (as we do for earn in `getEarnPosition`) and in library only extend interface with 

```
    public pnl: BigNumber,
    public totalEarnings: BigNumber,
```

Comment (zerotucks) [reply packages/dma-library/src/types/ajna/ajna-position.ts:135]: Yes. I had a chat with Damian about doing that way. But, we thought perhaps keeping the constructor lean and not having to update their usage elsewhere might be simpler.

File: packages/dma-library/src/types/ajna/ajna-position.ts (hunk 1 above)
Comment (zerotucks) [reply packages/dma-library/src/types/ajna/ajna-position.ts]: My bad. Copied the wrong one like a muppet 🤦‍♂️

Comment (paszkowskiDamian) [reply packages/dma-library/src/types/ajna/ajna-position.ts]: 🫂

**Review by paszkowskiDamian — APPROVED** [review]

**Pull request merged**